src/kartograf/filters/ring_changes.py

```diff
@@ -0,0 +1,73 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def filter_ringsize_changes(molA, molB,
+                            mapping: dict[int, int]) -> dict[int, int]:
+    """Prevents mutating the size of rings in the mapping"""
+    riA = molA.GetRingInfo()
+    riB = molB.GetRingInfo()
+
+    filtered_mapping = {}
+
+    for i, j in mapping.items():
+        at_i = molA.GetAtomWithIdx(i)
+        at_j = molB.GetAtomWithIdx(j)
+
+        # AtomRingSizes gives as tuple of ring sizes that this atom is part of
+        rs_i = set(riA.AtomRingSizes(at_i))
+        rs_j = set(riB.AtomRingSizes(at_j))
+
+        # if there's any intersection in ring size, we're ok
+        if not (rs_i & rs_j):
```

Comment:
empty sets (if both are not in a ring) cause a problem here which lead to unmapping.

```suggestion
        if not (rs_i & rs_j or len(rs_i) == len(rs_j) == 0):
```